lib/msf/core/exploit/remote/smb/relay/ntlm/server_client.rb

```diff
@@ -99,7 +99,7 @@ def do_session_setup_smb2(request, session)
       response.smb2_header.nt_status = relay_result.nt_status.value
       if relay_result.nt_status == ::WindowsError::NTStatus::STATUS_MORE_PROCESSING_REQUIRED
         response.smb2_header.nt_status = ::WindowsError::NTStatus::STATUS_MORE_PROCESSING_REQUIRED.value
-        response.buffer = relay_result.message.serialize
+        response.buffer = relay_result.message.serialize if relay_result.message
```

Comment:
I'm not sure why the NTLM type 2 message is not wrapped with GSS-API. I tried to do it but it fails, the server simply reset the connection. If anyone has any idea why?
```ruby
response.buffer = ::RubySMB::Gss.gss_type2(relay_result.message.serialize) if relay_result.message
```